File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumTipoMotivoPedido.js
var EnumTipoMotivoPedidoHelper = function () {
    this.Todos = "";
    this.LancamentoPedido = 0;
    this.AprovacaoPedido = 1;
    this.RejeicaoPedido = 2;
};

EnumTipoMotivoPedidoHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Lançamento Pedido", value: this.LancamentoPedido },
            { text:  "Aprovação Pedido", value: this.AprovacaoPedido },
            { text:   "Rejeição Pedido", value: this.RejeicaoPedido }
        ];
    },
    obterOpcoesPesquisa: function () {
        return [{ text: "Todos", value: this.Todos }].concat(this.obterOpcoes());
    }
}

var EnumTipoMotivoPedido = Object.freeze(new EnumTipoMotivoPedidoHelper());

File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumTipoMotivoPedido.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum EnumTipoMotivoPedido
    {
        LancamentoPedido = 0,
        AprovacaoPedido = 1,
        RejeicaoPedido = 2
    }

    public static class TipoMotivoPedidoHelper
    {
        public static string ObterDescricao(this EnumTipoMotivoPedido tipo)
        {
            switch (tipo)
            {
                case EnumTipoMotivoPedido.LancamentoPedido: return "Lançamento de Pedido";
                case EnumTipoMotivoPedido.AprovacaoPedido: return "Aprovação de Pedido";
                case EnumTipoMotivoPedido.RejeicaoPedido: return "Rejeição de Pedido";
                default: return string.Empty;
            }
        }
    }
}
